packages/cxl-ui/src/components/cxl-jw-player/mixins/TranscriptMixin.js

```diff
@@ -0,0 +1,191 @@
+
+    async _getTracks() {
+      const tracks = [];
+
+      const captions = await this._getCaptions();
+
+      if (captions.length) {
+        const chapters = [...[{ data: { start: 0, text: '' } }], ...(await this._getChapters())];
+
+        chapters.forEach((chapter, index) => {
+          tracks.push({ ...chapter, ...{ isChapter: true } });
+          tracks.push(...this._getCaptionsInChapter(chapters, captions, index));
+        });
+      }
+
+      return tracks;
+    }
+
+    _onCaptionClick(e) {
+      const index = Number(e.currentTarget.dataset.index);
+      this._jwPlayer.seek(this._tracks[index].data.start / 1000);
+    }
+
+    _onTimeListener(event) {
+      super._onTimeListener(event);
+
+      const position = event.position * 1000; // Convert to milliseconds
+
+      this._tracks.forEach(({ data: { end, start } }, index) => {
+        if (start <= position && end >= position) {
+          if (this.shouldScroll) {
+            const el = this.renderRoot.querySelector(`[data-index="${index}"]`);
+            if (el) {
+              el.scrollIntoView(true);
+            }
+          }
+
+          this._currentTrack = index;
+        }
+      });
+    }
+
+    _search() {
+      this._mark.unmark();
+
+      if (this._searchElement.value.length >= this.minimumSearchLength) {
+        this._isSearchMinimumLength = true;
+
+        this._mark.mark(this._searchElement.value, {
+          done: (total) => {
+            this._matches = total;
+          },
+          separateWordSearch: false,
+        });
+      } else {
+        this._isSearchMinimumLength = false;
+      }
+    }
+
+    async _setup() {
+      await super._setup();
+
+      this._setupTranscript();
+
+      this._jwPlayer.on('playlistItem', this._setupTranscript.bind(this));
+    }
+
+    async _setupTranscript() {
+      if (!this._jwPlayer) return;
+
+      this._tracks = [];
+
+      if (this.captions) {
+        this._tracks = await this._getTracks();
+      }
+
+      if (this._tracks.length) {
+        this.captions = true;
+
+        // Make sure the DOM is up to date
+        await this.updateComplete;
+
+        this._mark = new Mark(this.renderRoot.querySelectorAll('.captions h2, .captions span'));
+
+        this._jwPlayer.addButton(
+          `<svg class="jw-player-button" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" preserveAspectRatio="xMidYMid meet" aria-hidden="true" viewBox="0 0 1000 1000"><g><path d="M662 603l131 131c16 16 16 42 0 59-16 16-43 16-59 0l-131-131C562 691 512 708 458 708c-138 0-250-112-250-250 0-138 112-250 250-250 138 0 250 112 250 250 0 54-17 104-46 145zM458 646c104 0 188-84 188-188S562 271 458 271 271 355 271 458s84 188 187 188z"></path></g></svg>`,
+          'Transcript',
+          this._toggleTranscript.bind(this),
+          'toggle-transcript'
+        );
+      } else {
+        this.captions = false;
+        this._jwPlayer.removeButton('toggle-transcript');
+      }
```

Comment:
I don't see this button anywhere :(

Reply:
It's only available for videos with captions. It's the magnifying glass at the bottom right hand corner of the video.

![deploy-preview-433--conversionxl-aybolit netlify app__path=_story_cxl-ui-cxl-jw-player--default](https://github.com/user-attachments/assets/3800842e-4699-4fbf-9611-ffecc8f31368)